nni/compression/pytorch/compressor.py

```diff
@@ -615,6 +615,8 @@ def __init__(self, model, config_list, optimizer=None, dummy_input=None):
                 # we still need to deal with the old_bias when it occurs
                 if hasattr(wrapper.module, "old_bias"):
                     self.optimizer.add_param_group({"params": getattr(wrapper.module, "old_bias")})
+        else:
+            _logger.warning("Quantization step may not happen since optimizer is not set. Suggest passing an optimizer to Quantizer.")
```

Comment:
Not all quantizer needs an optimizer, suggest removing the default value `None` of `optimizer` in `QAT` and others, and move the warning in these quantizers with judging `isinstance(optimizer, Optimizer)`.

Reply:
Good point. Observer quantizer currently does not need optimizer

Reply:
Done.